packages/erk-shared/src/erk_shared/context/helpers.py

```diff
@@ -0,0 +1,328 @@
+
+
+def require_issues(ctx: click.Context) -> GitHubIssues:
+    """Get GitHub Issues from context, exiting with error if not initialized.
+
+    Uses LBYL pattern to check context before accessing. If context is not
+    initialized (ctx.obj is None), prints error to stderr and exits with code 1.
+
+    Args:
+        ctx: Click context (must have ErkContext in ctx.obj)
+
+    Returns:
+        GitHubIssues instance from context
+
+    Raises:
+        SystemExit: If context not initialized (exits with code 1)
+
+    Example:
+        >>> @click.command()
+        >>> @click.pass_context
+        >>> def my_command(ctx: click.Context) -> None:
+        ...     issues = require_issues(ctx)
+        ...     issues.add_comment(repo_root, issue_number, body)
+    """
+    from erk_shared.context.context import ErkContext
+
+    if ctx.obj is None:
+        click.echo("Error: Context not initialized", err=True)
+        raise SystemExit(1)
+
+    # Use isinstance for ErkContext, fall back to hasattr for duck typing in tests
+    if isinstance(ctx.obj, ErkContext):
+        return ctx.obj.issues
+    if hasattr(ctx.obj, "issues"):
+        return ctx.obj.issues
+    if hasattr(ctx.obj, "github_issues"):
+        return ctx.obj.github_issues
+
+    click.echo("Error: Context missing issues", err=True)
+    raise SystemExit(1)
+
+
+def require_repo_root(ctx: click.Context) -> Path:
+    """Get repo root from context, exiting with error if not initialized.
+
+    Uses LBYL pattern to check context before accessing.
+
+    Args:
+        ctx: Click context (must have ErkContext in ctx.obj)
+
+    Returns:
+        Path to repository root
+
+    Raises:
+        SystemExit: If context not initialized (exits with code 1)
+
+    Example:
+        >>> @click.command()
+        >>> @click.pass_context
+        >>> def my_command(ctx: click.Context) -> None:
+        ...     repo_root = require_repo_root(ctx)
+        ...     issues = require_issues(ctx)
+        ...     issues.create_issue(repo_root, title, body, labels)
+    """
+    from erk_shared.context.context import ErkContext
+
+    if ctx.obj is None:
+        click.echo("Error: Context not initialized", err=True)
+        raise SystemExit(1)
+
+    # Use isinstance for ErkContext, fall back to hasattr for duck typing in tests
+    if isinstance(ctx.obj, ErkContext):
+        repo = ctx.obj.repo
+        if isinstance(repo, NoRepoSentinel):
+            click.echo("Error: Not in a git repository", err=True)
+            raise SystemExit(1)
+        return repo.root
+    if hasattr(ctx.obj, "repo"):
+        repo = ctx.obj.repo
+        if isinstance(repo, NoRepoSentinel):
+            click.echo("Error: Not in a git repository", err=True)
+            raise SystemExit(1)
+        return repo.root
+
+    if hasattr(ctx.obj, "repo_root"):
+        return ctx.obj.repo_root
+
+    click.echo("Error: Context missing repo", err=True)
+    raise SystemExit(1)
```

Comment:
hard fail if not ErkContext. simplify

Reply:
Resolved via /erk:pr-address at 2025-12-10 06:01 AM EST - Simplified to hard fail if not ErkContext

_Addressed via `/erk:pr-address` at 2025-12-10 06:01 EST_